lib/recipe/widgets/recipe_details_screen.dart

```diff
@@ -19,51 +19,100 @@ class RecipeDetailsScreen extends StatelessWidget {
     return BlocBuilder<RecipeBloc, RecipeState>(builder: (context, state) {
       return Scaffold(
         appBar: AppBar(title: const Text('Recipe details')),
-        body: RecipeDetails(recipeIndex),
-        floatingActionButton: FloatingActionButton.extended(
-            heroTag: null,
-            label: assignedFoodItemId == null
-                ? const Text("Save Recipe")
-                : const Text("Choose Recipe"),
-            icon: assignedFoodItemId == null
-                ? const Icon(Icons.save_sharp)
-                : const Icon(Icons.check),
-            onPressed: () {
-              if (assignedFoodItemId == null) {
-                context.read<RecipeBloc>().add(RecipeStatusChanged(
-                    recipe: state.recipes[recipeIndex], recipeSaved: true));
-                context.read<RecipeBloc>().add(const RecipesSaved());
-                Navigator.pop(context);
-              } else {
-                context.read<CollectionBloc>().add(FoodItemRecipeChanged(
-                    foodItemId: assignedFoodItemId!,
-                    foodItemRecipe: state.recipes[recipeIndex]));
-                context.read<RecipeBloc>().add(const RecipesSaved());
-                Navigator.pop(context);
-                Navigator.pop(context);
-              }
-            }),
+        body:
+            RecipeDetails(recipeIndex, assignedFoodItemId: assignedFoodItemId),
+        floatingActionButton: Visibility(
+          visible:
+              !state.recipes[recipeIndex].saved || assignedFoodItemId != null,
+          child: FloatingActionButton.extended(
+              heroTag: null,
+              label: assignedFoodItemId == null
```

Comment:
Similarly for these null checks — a well-named bool would make it much easier to understand when reading this for the first time imo.